Importer/CsvImporter.php

```diff
@@ -23,88 +27,241 @@ class CsvImporter extends AbstractImporter
-     * @param string                   $delimiter
+     * @var ConfigurationInterface
+     */
+    protected $configuration;
+
+    /**
+     * @var EntityManager
+     */
+    protected $entityManager;
+
+    /**
+     * @var ImporterErrorHandlerInterface[]
+     */
+    protected $importerErrorHandlers = [];
+
+    /**
+     * @var InterpreterInterface
+     */
+    protected $interpreter;
+
+    /**
+     * @var Log
+     */
+    protected $log;
+
+    /**
+     * @var bool
+     */
+    protected $headers;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param InterpreterInterface   $interpreter
+     * @param EntityManager          $entityManager
+     * @param CsvParser              $parser
+     * @param string                 $delimiter
+     * @param Log                    $log
+     * @param boolean                $headers
      */
     public function __construct(
         ConfigurationInterface $configuration,
         InterpreterInterface $interpreter,
         EntityManager $entityManager,
         CsvParser $parser,
-        $delimiter = ','
-    )
-    {
+        $delimiter = ',',
+        Log $log,
+        $headers = true
+    ) {
         $this->parser = $parser;
-        parent::__construct($configuration, $interpreter, $entityManager);
         $this->delimiter = $delimiter;
+        $this->configuration = $configuration;
+        $this->entityManager = $entityManager;
+        $this->interpreter = $interpreter;
+        $this->log = $log;
+        $this->headers = $headers;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasHeaders()
+    {
+        return $this->headers;
+    }
+
+    /**
+     * @param bool $headers
+     */
+    public function setHeaders($headers)
+    {
+        $this->headers = $headers;
+    }
+
+    /**
+     * @param string $delimiter
+     */
+    public function setDelimiter($delimiter)
+    {
+        $this->delimiter = $delimiter;
+    }
+
+    /**
+     * @param InterpreterErrorHandlerInterface $lineErrorHandler
+     */
+    public function registerInterpreterErrorHandler(InterpreterErrorHandlerInterface $lineErrorHandler)
+    {
+        $this->interpreter->registerErrorHandler($lineErrorHandler);
+    }
+
+    /**
+     * @param ImporterErrorHandlerInterface $fileErrorHandler
+     */
+    public function registerImporterErrorHandler(ImporterErrorHandlerInterface $fileErrorHandler)
+    {
+        $this->importerErrorHandlers[] = $fileErrorHandler;
+    }
+
+    /**
+     * @return ConfigurationInterface
+     */
+    public function getConfiguration()
+    {
+        return $this->configuration;
     }
 
     /**
      * @param string $filename
-     * @param bool   $hasHeaders
      * @param bool   $dryRun
-     *
-     * @return array|null|object[]
-     * @throws DatabaseException
      */
-    public function importFile($filename, $hasHeaders = true, $dryRun = false)
+    public function importFile($filename, $dryRun = false)
```

Comment:
Actually I really liked the idea of `CsvFileImporter`. I find this very clear, and at the same it shows me that If I want to create a `CsvImageImporter`, then I have to manipulate the `image` to `csv` and it will work as expected.